### PR TITLE
Rearrange CLI modules

### PR DESCRIPTION
**Pull request by dcoutts**
Split the command types out of Cardano.CLI.Shelley.Parsers

Move the Shelley run commands into their own modules
    
We were mostly following this pattern already. This just makes it
consistent. So the module structure exactly follows the CLI
command hierarchy.

### CODE REVIEW

File: cardano-cli/src/Cardano/CLI/Shelley/KeyGen.hs [hunk 1]
```diff
@@ -0,0 +1,38 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE StrictData #-}
+
+{-# OPTIONS_GHC -Wno-all-missed-specialisations #-}
+{-# OPTIONS_GHC -Wno-unticked-promoted-constructors #-}
```
Comment (intricate) [inline cardano-cli/src/Cardano/CLI/Shelley/KeyGen.hs:9]: At a glance, I think most of these pragmas are redundant:

```suggestion
{-# LANGUAGE DataKinds #-}

{-# OPTIONS_GHC -Wno-unticked-promoted-constructors #-}
```

File: cardano-cli/src/Cardano/CLI/Shelley/Run.hs [hunk 2]
```diff
@@ -10,40 +10,22 @@ module Cardano.CLI.Shelley.Run
...
 import           Cardano.CLI.Ops (CliError (..))
 import           Cardano.CLI.Shelley.Parsers
 
 import           Cardano.CLI.Shelley.Run.Address
+import           Cardano.CLI.Shelley.Run.StakeAddress
+import           Cardano.CLI.Shelley.Run.Transaction
+import           Cardano.CLI.Shelley.Run.Node
+import           Cardano.CLI.Shelley.Run.Pool
+import           Cardano.CLI.Shelley.Run.Query
+                                         -- Block, System, DevOps
```
Comment (intricate) [inline cardano-cli/src/Cardano/CLI/Shelley/Run.hs:26]:
```suggestion
```

File: cardano-cli/src/Cardano/CLI/Shelley/Parsers.hs [hunk 3]
```diff
@@ -2,264 +2,35 @@ module Cardano.CLI.Shelley.Parsers
   ( -- * CLI command parser
     parseShelleyCommands
 
-    -- * CLI command types
-  , ShelleyCommand (..)
```
Comment (Jimbo4350) [inline cardano-cli/src/Cardano/CLI/Shelley/Parsers.hs:6]: :+1:

File: cardano-cli/src/Cardano/CLI/Shelley/Run.hs [hunk 4]
```diff
@@ -63,262 +45,15 @@ runShelleyClientCommand (DevOpsCmd       cmd) = runDevOpsCmd       cmd
...
-
-
-runPoolCmd :: PoolCmd -> ExceptT CliError IO ()
-runPoolCmd (PoolRegistrationCert sPvkey vrfVkey pldg pCost pMrgn rwdVerFp ownerVerFps relays outfp) =
-  runStakePoolRegistrationCert sPvkey vrfVkey pldg pCost pMrgn rwdVerFp ownerVerFps relays outfp
-runPoolCmd (PoolRetirmentCert sPvkeyFp retireEpoch outfp) =
-  runStakePoolRetirementCert sPvkeyFp retireEpoch outfp
-runPoolCmd cmd = liftIO $ putStrLn $ "runPoolCmd: " ++ show cmd
 
+--TODO: if you fill any of these in, move them into their own modules first!
```
Comment (Jimbo4350) [inline cardano-cli/src/Cardano/CLI/Shelley/Run.hs:49]: :+1:

File: cardano-cli/src/Cardano/CLI/Shelley/Run/StakeAddress.hs [hunk 5]
```diff
@@ -0,0 +1,71 @@
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
```
Comment (intricate) [inline cardano-cli/src/Cardano/CLI/Shelley/Run/StakeAddress.hs:5]:
```suggestion
```

File: cardano-cli/src/Cardano/CLI/Shelley/Run/Pool.hs [hunk 6]
```diff
@@ -0,0 +1,126 @@
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
```
Comment (intricate) [inline cardano-cli/src/Cardano/CLI/Shelley/Run/Pool.hs:5]:
```suggestion
```

**Comment by dcoutts** [comment]
@intricate you're almost certainly right about the language pragmas.

Mind if I prune those as another PR, since this one that moves things about will conflict with things, so I'd rather get it in, and we can do more localised changes more easily.

**Comment by dcoutts** [comment]
bors merge

**Comment by intricate** [comment]
> @intricate you're almost certainly right about the language pragmas.
> 
> Mind if I prune those as another PR, since this one that moves things about will conflict with things, so I'd rather get it in, and we can do more localised changes more easily.

Sure! I just wanted to make sure I noted those things.

**Pull request merged**